modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/snapshot/incoming/IncomingSnapshotCopier.java

```diff
@@ -113,8 +147,174 @@ public SnapshotReader getReader() {
         return new IncomingSnapshotReader(snapshotMeta);
     }
 
-    @Override
-    public void close() throws IOException {
-        threadPool.shutdownNow();
+    private void start0() {
+        if (!cancelBusyLock.enterBusy()) {
+            return;
+        }
+
+        try {
+            partitionSnapshotStorage.partition().reCreatePartition()
+                    .whenComplete((unused, throwable) -> {
```

Comment:
Why do you use `sync` version of the method? What thread will execute this code?

Reply:
Tried to fix it.